styleguides/javascript/react.md

```diff
@@ -0,0 +1,622 @@
+
+The style guide assumes you are using a miniumum of ES6 (which you should always be using) & babel to transpile down for your target environment (if needed).
+
+We make use of [ESLint](https://eslint.org/) across the board for our JS projects to ensure conformance to the below styles, a custom eslint preset is coming soon for our modified version of the Airbnb style guide although for now the Airbnb React preset can be used with a few rules turned off.
+
+This style guide is mostly based on the standards that are currently prevalent in JavaScript, although some conventions (i.e async/await or static class fields) may still be included or prohibited on a case-by-case basis. Currently, anything prior to stage 3 is not included nor recommended in this guide.
+
+## Table of Contents
+
+  1. [Basic Rules](#basic-rules)
+  1. [Class vs `React.createClass` vs stateless](#class-vs-reactcreateclass-vs-stateless)
+  1. [Mixins](#mixins)
+  1. [Naming](#naming)
+  1. [Declaration](#declaration)
+  1. [Alignment](#alignment)
+  1. [Quotes](#quotes)
+  1. [Spacing](#spacing)
+  1. [Props](#props)
+  1. [Refs](#refs)
+  1. [Parentheses](#parentheses)
+  1. [Tags](#tags)
+  1. [Methods](#methods)
+  1. [Ordering](#ordering)
+
+## Basic Rules
+
+  - Only include one React component per file.
+    - However, multiple [Stateless, or Pure, Components](https://facebook.github.io/react/docs/reusable-components.html#stateless-functions) are allowed per file. eslint: [`react/no-multi-comp`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-multi-comp.md#ignorestateless).
+  - Always use JSX syntax.
+  - Do not use `React.createElement` unless you’re initializing the app from a file that is not JSX.
+
+## Class vs `React.createClass` vs stateless
+
+  - If you have internal state and/or refs, prefer `class extends React.Component` over `React.createClass`. eslint: [`react/prefer-es6-class`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-es6-class.md) [`react/prefer-stateless-function`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)
+
+    ```jsx
+    // bad
+    const Listing = React.createClass({
+      // ...
+      render() {
+        return <div>{this.state.hello}</div>;
+      }
+    });
+
+    // good
+    class Listing extends React.Component {
+      // ...
+      render() {
+        return <div>{this.state.hello}</div>;
+      }
+    }
+    ```
+
+    And if you don’t have state or refs, prefer normal functions (not arrow functions) over classes:
+
+    ```jsx
+    // bad
+    class Listing extends React.Component {
+      render() {
+        return <div>{this.props.hello}</div>;
+      }
+    }
+
+    // bad (relying on function name inference is discouraged)
+    const Listing = ({ hello }) => (
+      <div>{hello}</div>
+    );
+
+    // good
+    function Listing({ hello }) {
+      return <div>{hello}</div>;
+    }
+    ```
```

Comment:
It's to make your code easier to understand.

In React you have the concept of presentational and container components, presentational components are always represented as functions because they just care about how things look so will rarely ever have state, this means you can write less, more concise code, it also allows you easily see a glance what's presentational and what's not.

I feel writing more code now to avoid having to possibly write more code in the future isn't necessary and I'm a big fan of stateless components for the ability to write cleaner code.

Reply:
Okay, sure. That's reasonable. 👍